### PR TITLE
Don't fail bud --layers on dangling symlink

### PR DESCRIPTION
**Pull request by umohnani8**
If there is a dangling symlink when building with --layers,
don't fail and continue the build process.
It would fail because it would try to stat the file to check
whether it has been modified since the previous build. And since
it would be a dangling symlink, the stat would fail hence causing
the build to fail.

Fixes https://github.com/containers/buildah/issues/1219

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@rhatdan @TomSweeneyRedHat PTAL

File: imagebuildah/build.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/containers/storage"
 	"github.com/containers/storage/pkg/archive"
 	docker "github.com/fsouza/go-dockerclient"
-	"github.com/opencontainers/image-spec/specs-go/v1"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
```
Comment (rhatdan) [inline imagebuildah/build.go]: Why is this necessary?

**Comment by rhatdan** [comment]
Other then the question LGTM

File: imagebuildah/build.go (hunk 1 above)
Comment (umohnani8) [reply imagebuildah/build.go]: vscode did that automatically lol, fixed.

**Comment by rhatdan** [comment]
lgtm
@nalind @vrothberg @giuseppe @TomSweeneyRedHat PTAL

File: imagebuildah/chroot_symlink.go [hunk 2]
```diff
@@ -19,6 +19,10 @@ const (
 	maxSymlinksResolved    = 40
 )
 
+var (
+	errDanglingSymlink = errors.New("error evaluating dangling symlink")
```
Comment (nalind) [inline imagebuildah/chroot_symlink.go]: Creating this with the `github.com/pkg/errors` package adds a traceback that points to this line whenever we return the error.  Can you change this to be created with the standard library's `errors` package, either by importing it under a different name or making it the sole definition in an `errors.go` that uses the standard `errors` package?

File: imagebuildah/chroot_symlink.go [hunk 3]
```diff
@@ -154,6 +158,9 @@ func modTimeIsGreater(rootdir, path string, historyTime string) (bool, error) {
 		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
 			// Evaluate any symlink that occurs to get updated modified information
 			resolvedPath, err := filepath.EvalSymlinks(path)
+			if err != nil && os.IsNotExist(err) {
+				return errDanglingSymlink
```
Comment (nalind) [inline imagebuildah/chroot_symlink.go]: If `errDanglingSymlink` becomes a standard-library-produced `error`, this will probably want to be wrapped with `errors.Wrapf()`...

File: imagebuildah/chroot_symlink.go [hunk 4]
```diff
@@ -169,7 +176,12 @@ func modTimeIsGreater(rootdir, path string, historyTime string) (bool, error) {
 		}
 		return nil
 	})
+
 	if err != nil {
+		// if error is due to dangling symlink, ignore error and return nil
+		if err == errDanglingSymlink {
```
Comment (nalind) [inline imagebuildah/chroot_symlink.go]: ... and this will need to compare with `errors.Cause(err)`.

File: imagebuildah/chroot_symlink.go (hunk 2 above)
Comment (umohnani8) [reply imagebuildah/chroot_symlink.go]: Fixed. Created a new file called errors.go and used the standard library's error package.

File: imagebuildah/chroot_symlink.go (hunk 3 above)
Comment (umohnani8) [reply imagebuildah/chroot_symlink.go]: Fixed.

File: imagebuildah/chroot_symlink.go (hunk 4 above)
Comment (umohnani8) [reply imagebuildah/chroot_symlink.go]: Fixed

**Comment by umohnani8** [comment]
@nalind fixed the issues, PTAL

**Comment by rhatdan** [comment]
@rh-atomic-bot r+
